### ISSUE TITLE
Training code and pretrained weights? 

### ISSUE THREAD

**Issue by zj-jayzhang** [UCSC-VLAA/AdvXL#3]
Hey, cool work! Are you guys planning to release the training code and pre-trained model weights? It would make it easier for people to make fair comparisons on more settings.

**Comment by HashmatShadab** [comment]
Yes. That would be of great help. Hope they release the pretrained weights/ training code soon.

**Comment by HashmatShadab** [comment]
Kindly provide any update on this issue

**Comment by zw615** [comment]
@HashmatShadab @zj-jayzhang Hi, thanks for your interest in our work! I have just released ViT-L and ViT-H adversarially pre-trained CLIP model weights at https://github.com/zw615/Double_Visual_Defense. Apologize for the delay. The model weights was under some internal review process, which took a much longer time than expected. Let me know if you have any other questions, and I shall be happy to help.

As for the training code, it is implemented in JAX and needs some extra cleaning. I am working on it, and am planning to release them in the near future.

**Comment by zw615** [comment]
As for the training code, it is implemented in JAX and needs some extra cleaning. I am working on it, and am planning to release them in the near future.

**Comment by HashmatShadab** [comment]
@zw615  Thank you for releasing the ViT-L and ViT-H adversarially pre-trained CLIP model weights, even after an extensive internal review process that spanned nearly a year. Due to limited computational resources, I focused on analysis of previously available weights to demonstrate improvements in robustness within Multi-Modal Large Language Models. For more details, you can explore the [project page](https://hashmatshadab.github.io/Robust-LLaVA/) for our [technical report](https://arxiv.org/abs/2502.01576), [codebase](https://github.com/HashmatShadab/Robust-LLaVA) and the [Robust MLLM weights](https://drive.google.com/drive/folders/1mt5zbiWi_ZYNJyDCpJ33Zc3AFZ9NxaLr). 

I appreciate the release of the model weights and your insightful work in [Double Visual Defense](https://github.com/zw615/Double_Visual_Defense), which explores a similar direction to my research. It’s great to see efforts in incorporating robust backbones into MLLMs, and I look forward to potential discussions in this space.
